go_sequencer/txprocessor/txprocessor.go

```diff
@@ -1242,3 +1243,49 @@ func (txProcessor *TxProcessor) applyExit(coordIdxsMap map[common.TokenID]common
 
 	return exitAccount, false, nil
 }
+
+// computeEffectiveAmounts checks that the L1Tx data is correct
+func (txProcessor *TxProcessor) computeEffectiveAmounts(tx *common.L1Tx) {
+	tx.EffectiveAmount = tx.Amount
+	tx.EffectiveDepositAmount = tx.DepositAmount
+
```

Comment:
@kaiden-araki Regarding checks for handling the coordinator transactions. We'll need to add checks for the same  - to check if txs is originated by user or by coordinator. 

```
if !tx.UserOrigin {
		// case where the L1Tx is generated by the Coordinator
		tx.EffectiveAmount = big.NewInt(0)
		tx.EffectiveDepositAmount = big.NewInt(0)
		return
	}
	
```

Reply:
Yeah, let's check coordinator is necessary or not.